server/server.go

```diff
@@ -151,15 +182,21 @@ func NewServer(cfg *config.Config, driver IDriver) (*Server, error) {
 	}
 
 	var err error
-	if cfg.Socket != "" {
-		if s.listener, err = net.Listen("unix", cfg.Socket); err == nil {
-			log.Infof("Server is running MySQL Protocol through Socket [%s]", cfg.Socket)
-		}
-	} else {
+
+	if s.cfg.Host != "" && s.cfg.Port != 0 {
 		addr := fmt.Sprintf("%s:%d", s.cfg.Host, s.cfg.Port)
 		if s.listener, err = net.Listen("tcp", addr); err == nil {
 			log.Infof("Server is running MySQL Protocol at [%s]", addr)
+			if cfg.Socket != "" {
+				go s.forwardUnixSocketToTCP(cfg.Socket, addr) // listen on both
```

Comment:
We can listen to the socket here, only handle connections in goroutine.

Reply:
I like the idea, but if I inline and remove the goroutine, it blocks (the for loop below with `socket.Accept()` needs to be run async).  Ideas?

Reply:
I think I made a mistake in my testing; it is blocking on the socket regardless :(

Reply:
Listen doesn't block, `Accept` in a goroutine.